### PR TITLE
Initial work on AWS public cloud support for ubernetes lite

### PR DESCRIPTION
**Pull request by tayzlor**
todo - 

- [ ] getting started docs
- [ ] currently master nodes are distributed across 3 AZs - though i'm not sure this is currently recommended / supported?
- [ ] ELB(?) or route53 dns(?) for Master instances (so you can connect through load balancer rather than direct to IP address of master 0)

it's potentially worth creating a follow up to abstract out all the 'certs' to a local module which could be re-used across providers here.

### CODE REVIEW

File: terraform/aws/public-cloud/etcd_discovery_url.txt [hunk 1]
```diff
@@ -0,0 +1 @@
+https://discovery.etcd.io/2bde7078922de146c40383145a581978
```
Comment (wallies) [inline terraform/aws/public-cloud/etcd_discovery_url.txt]: can we leave this empty

**Comment by tayzlor** [comment]
Some initial testing of this looks like a prob with certs registering public ips instead of private ones - 

```
Mar 31 17:39:00 ip-10-0-1-195.eu-west-1.compute.internal kubelet-wrapper[1561]: E0331 16:39:00.140088    1561 event.go:202] Unable to write event: 'Post https://10.0.3.81/api/v1/namespaces/default/events: x509: certificate is valid for 10.3.0.1, 52.49.74.118, not 10.0.3.81' (may retry after sleeping)```

we should be able to use the private ips in AWS for the certs

**Comment by tayzlor** [comment]
This PR also needs to handle setting ```--cloud-provider``` and ```node-labels``` properly via the kubelet. The failure domain/zone labels etc are not coming through for the nodes yet

http://kubernetes.io/docs/admin/kubelet/

**Comment by tayzlor** [comment]
https://github.com/Capgemini/kubeform/pull/64/commits/51bcd65260df05c18d6181cc0687f1b0bb4b28d9 addresses --cloud-provider support + correct node labels for the zone/regions

**Comment by tayzlor** [comment]
merging as is will create follow-up PRs to address further issues

**Pull request merged**